### DISCUSSION TITLE
Older plants disappear when new plant is added

### DISCUSSION THREAD

**Discussion by Reka27** [neuefische/questions#851] [Web] [answered]
Hi all, I am currently working on edit-plant. when i try to  add a new plant other already created plants disappear. I think this problem is due to navigating to the formpage reloads the browser.  I dont know how to fix it. Any help would be highly appreciated.
https://github.com/MarenOelixtown/plant-care-app/tree/edit-plant/components

**Comment by NikRoe** [comment, marked as answer]
Hi Reka,

your assumption is absolutely correct. Whenever you navigate to the page where you can add new plants, you lose all previously added plants and revert to the initial state.

The issue arises from the way you navigate to the form page. Currently, you have a button with an onClick event that triggers a handleButtonClick function. This function uses window.location.assign to redirect the user, which inevitably causes your application to reload. As a result, your application's state is reset.

Using a button to trigger navigation is not the best approach if the destination page is known beforehand. It is only advisable if some calculations are needed before determining the redirection path.

In your scenario, since the destination is predetermined, you should use the Link component provided by Next.js. This will enable smoother transitions between pages without reloading the application and resetting its state.

In your `MyPlants` component, you should change your StyledButton to be a StyledLink instead:
```js
const StyledLink = styled(Link)`
  background: none;
  cursor: pointer;
  padding: 5px;
  margin-top: 10px;
`;
```
> you might need to adjust some of the styling, since a link has different default styling than a button

When you call this component in your JSX it should look like this:
```js
// ...
<StyledLink href="/plantformpage">Add a new plant</StyledLink>
/// ...
```

I hope this already does the trick 🤞

**Reply by Reka27** [reply]
Thanks Niklas, Followed your suggestion and it worked and the problem is solved
